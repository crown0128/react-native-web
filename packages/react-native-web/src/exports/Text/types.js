/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import type { ColorValue, GenericStyleProp, LayoutEvent } from '../../types';
import type { ViewStyle } from '../View/types';

type FontWeightValue =
  | 'normal'
  | 'bold'
  | '100'
  | '200'
  | '300'
  | '400'
  | '500'
  | '600'
  | '700'
  | '800'
  | '900';

export type TextStyle = {
  ...ViewStyle,
  color?: ColorValue,
  fontFamily?: string,
  fontFeatureSettings?: string,
  fontSize?: number | string,
  fontStyle?: 'italic' | 'normal',
  fontWeight?: FontWeightValue,
  fontVariant?: $ReadOnlyArray<
    'small-caps' | 'oldstyle-nums' | 'lining-nums' | 'tabular-nums' | 'proportional-nums'
  >,
  letterSpacing?: number | string,
  lineHeight?: number | string,
  textAlign?: 'center' | 'end' | 'inherit' | 'justify' | 'justify-all' | 'left' | 'right' | 'start',
  textAlignVertical?: string,
  textDecorationColor?: ColorValue,
  textDecorationLine?: 'none' | 'underline' | 'line-through' | 'underline line-through',
  textDecorationStyle?: 'solid' | 'double' | 'dotted' | 'dashed',
  textIndent?: number | string,
  textOverflow?: string,
  textRendering?: 'auto' | 'geometricPrecision' | 'optimizeLegibility' | 'optimizeSpeed',
  textShadowColor?: ColorValue,
  textShadowOffset?: {| width?: number, height?: number |},
  textShadowRadius?: number,
  textTransform?: 'capitalize' | 'lowercase' | 'none' | 'uppercase',
  unicodeBidi?: 'normal' | 'bidi-override' | 'embed' | 'isolate' | 'isolate-override' | 'plaintext',
  whiteSpace?: string,
  wordBreak?: 'normal' | 'break-all' | 'break-word' | 'keep-all',
  wordWrap?: string,
  writingDirection?: 'auto' | 'ltr' | 'rtl',
  /* @platform web */
  MozOsxFontSmoothing?: string,
  WebkitFontSmoothing?: string
};

export type TextProps = {
  accessibilityLabel?: string,
  accessibilityLiveRegion?: 'none' | 'polite' | 'assertive',
  accessibilityRelationship?: {
    activedescendant?: ?string,
    controls?: ?string,
    describedby?: ?string,
    details?: ?string,
    haspopup?: ?string,
    labelledby?: ?string,
    owns?: ?string
  },
  accessibilityRole?:
    | 'button'
    | 'header'
    | 'heading'
    | 'label'
    | 'link'
    | 'listitem'
    | 'none'
    | 'text',
  accessibilityState?: {
    busy?: ?boolean,
    checked?: ?boolean | 'mixed',
    disabled?: ?boolean,
    expanded?: ?boolean,
    grabbed?: ?boolean,
    hidden?: ?boolean,
    invalid?: ?boolean,
    pressed?: ?boolean,
    readonly?: ?boolean,
    required?: ?boolean,
    selected?: ?boolean
  },
  accessible?: boolean,
  children?: any,
  dir?: 'auto' | 'ltr' | 'rtl',
  forwardedRef?: any,
  importantForAccessibility?: 'auto' | 'yes' | 'no' | 'no-hide-descendants',
  nativeID?: string,
  numberOfLines?: number,
  onBlur?: (e: any) => void,
  onFocus?: (e: any) => void,
  onLayout?: (e: LayoutEvent) => void,
  onPress?: (e: any) => void,
  onMoveShouldSetResponder?: (e: any) => boolean,
  onMoveShouldSetResponderCapture?: (e: any) => boolean,
  onResponderEnd?: (e: any) => void,
  onResponderGrant?: (e: any) => void,
  onResponderMove?: (e: any) => void,
  onResponderReject?: (e: any) => void,
  onResponderRelease?: (e: any) => void,
  onResponderStart?: (e: any) => void,
  onResponderTerminate?: (e: any) => void,
  onResponderTerminationRequest?: (e: any) => void,
  onScrollShouldSetResponder?: (e: any) => boolean,
  onScrollShouldSetResponderCapture?: (e: any) => boolean,
  onSelectionChangeShouldSetResponder?: (e: any) => boolean,
  onSelectionChangeShouldSetResponderCapture?: (e: any) => boolean,
  onStartShouldSetResponder?: (e: any) => boolean,
  onStartShouldSetResponderCapture?: (e: any) => boolean,
  selectable?: boolean,
  style?: GenericStyleProp<TextStyle>,
  testID?: string,
  // unstable
  onContextMenu?: (e: any) => void,
  onKeyDown?: (e: any) => void,
  onKeyPress?: (e: any) => void,
  onKeyUp?: (e: any) => void,
  onMouseDown?: (e: any) => void,
  onMouseEnter?: (e: any) => void,
  onMouseLeave?: (e: any) => void,
  onMouseMove?: (e: any) => void,
  onMouseOver?: (e: any) => void,
  onMouseOut?: (e: any) => void,
  onMouseUp?: (e: any) => void,
  onTouchCancel?: (e: any) => void,
  onTouchCancelCapture?: (e: any) => void,
  onTouchEnd?: (e: any) => void,
  onTouchEndCapture?: (e: any) => void,
  onTouchMove?: (e: any) => void,
  onTouchMoveCapture?: (e: any) => void,
  onTouchStart?: (e: any) => void,
  onTouchStartCapture?: (e: any) => void,
  href?: string,
  itemID?: string,
  itemRef?: string,
  itemProp?: string,
  itemScope?: string,
  itemType?: string,
  rel?: string,
  target?: string,
  unstable_ariaSet?: Object,
  unstable_dataSet?: Object
};
