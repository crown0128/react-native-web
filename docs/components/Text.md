# Text

`Text` is component for displaying text. It supports style, basic touch
handling, and inherits typographic styles from ancestor elements. In a
divergence from React Native, components other than `Text` can be children of a
`Text` component.

The `Text` is unique relative to layout: child elements use text layout
(`inline-block`) rather than flexbox layout. This means that elements inside of
a `Text` are not rectangles, as they wrap when reaching the edge of their
container.

## Props

NOTE: `Text` will transfer all other props to the rendered HTML element.

**children** any

Child content.

**component** function or string

Default is `span`.

**numberOfLines** number.

Truncates the text with an ellipsis after this many lines.

**onPress** function

This function is called on press.

**style** style

+ `backgroundColor`
+ `color`
+ `direction`
+ `fontFamily`
+ `fontSize`
+ `fontStyle`
+ `fontWeight`
+ `letterSpacing`
+ `lineHeight`
+ `margin`
+ `padding`
+ `textAlign`
+ `textDecoration`
+ `textTransform`
+ `whiteSpace`
+ `wordWrap`

**testID** string

Used to locate this view in end-to-end tests.

## Examples

```js
import React, { Text } from 'react-native-web'

const { Component, PropTypes } = React

class PrettyText extends Component {
  static propTypes = {
    color: PropTypes.oneOf(['white', 'gray', 'red']),
    size: PropTypes.oneOf(['small', 'normal', 'large']),
    weight: PropTypes.oneOf(['light', 'normal', 'bold'])
  }

  static defaultProps = {
    color: 'gray',
    size: 'normal',
    weight: 'normal'
  }

  render() {
    const { color, size, style, weight, ...other } = this.props;

    return (
      <Text
        ...other
        style={{
          ...style,
          ...localStyle.color[color],
          ...localStyle.size[size],
          ...localStyle.weight[weight]
        }}
      />
    );
  }
}

const localStyle = {
  color: {
    white: { color: 'white' },
    gray: { color: 'gray' },
    red: { color: 'red' }
  },
  size: {
    small: { fontSize: '0.85rem', padding: '0.5rem' },
    normal: { fontSize: '1rem', padding: '0.75rem' },
    large: { fontSize: '1.5rem', padding: '1rem' }
  },
  weight: {
    light: { fontWeight: '300' },
    normal: { fontWeight: '400' },
    bold: { fontWeight: '700' }
  }
}
```
