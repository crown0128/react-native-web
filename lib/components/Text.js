import Component from './Component';
import {pickProps} from '../filterObjectProps';
import React, {PropTypes} from 'react';
import StylePropTypes from '../StylePropTypes';
import {ViewStylePropTypes} from './View';

const TextStyleDefaultProps = {
  alignItems: 'stretch', /* 1 */
  borderWidth: '0',
  borderStyle: 'solid',
  boxSizing: 'border-box', /* 2 */
  display: 'flex', /* 3 */
  flexBasis: 'auto', /* 1 */
  flexDirection: 'column', /* 1 */
  flexShrink: 0, /* 1 */
  listStyle: 'none',
  margin: '0',
  padding: '0',
  position: 'relative' /* 4 */
};

const TextStylePropTypes = {
  ...ViewStylePropTypes,
  ...StylePropTypes.TypographicPropTypes
};

class Text extends React.Component {
  static _getPropTypes() {
    return {
      className: PropTypes.string,
      element: PropTypes.oneOfType([
        PropTypes.string, PropTypes.func
      ]),
      style: PropTypes.shape(TextStylePropTypes)
    };
  }

  static _getDefaultProps() {
    return {
      className: '',
      element: 'div',
      style: {}
    };
  }

  render() {
    const { className, element, style, ...other } = this.props;
    const filteredStyle = pickProps(style, Object.keys(TextStylePropTypes));
    const mergedStyle = { ...TextStyleDefaultProps, ...filteredStyle };

    return (
      <Component
        {...other}
        className={`Text ${className}`}
        element={element}
        style={mergedStyle}
      />
    );
  }
}

Text.propTypes = Text._getPropTypes();
Text.defaultProps = Text._getDefaultProps();

export default Text;
export {TextStylePropTypes};
