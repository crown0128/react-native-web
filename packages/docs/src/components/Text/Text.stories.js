import PropTypes from 'prop-types';

const ofProps = () => {};

ofProps.propTypes = {
  '...ViewPropTypes': PropTypes.any,
  numberOfLines: PropTypes.number,
  onPress: PropTypes.func,
  selectable: PropTypes.bool
};

export default {
  title: 'Components|Text',
  includeStories: []
};

export { ofProps };

export { default as children } from './examples/Children';
export { default as numberOfLines } from './examples/NumberOfLines';
export { default as onPress } from './examples/OnPress';
export { default as selectable } from './examples/Selectable';
export { default as style } from './examples/Style';
