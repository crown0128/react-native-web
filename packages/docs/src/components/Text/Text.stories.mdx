import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';
import * as Stories from './examples';

<Meta title="Components|Text" />

# Text

Text is component for displaying text. It supports style, basic touch handling,
and inherits typographic styles from ancestor elements.

Text is unique relative to layout: child elements use text layout ("inline")
rather than flexbox layout. This means that elements inside of a Text are not
rectangles, as they wrap when reaching the edge of their container.

## Props

| Name                      | Type      | Default |
| ------------------------- | --------- | ------- |
| ...ViewProps              |           | |
| dir                       | ?string   | 'auto' |
| numberOfLines             | ?number   | |
| onPress                   | ?Function | |
| selectable                | ?boolean  | true |
| style                     | ?Style    | |

### children

Nested text components will inherit the typographic styles of their parents.

<Preview withSource='none'>
  <Story name="children">
    <Stories.children />
  </Story>
</Preview>

### numberOfLines

Truncates the text with an ellipsis after this many lines.

<Preview withSource='none'>
  <Story name="numberOfLines">
    <Stories.numberOfLines />
  </Story>
</Preview>

### onPress

Called when the `Text` is pressed.

<Preview withSource='none'>
  <Story name="onPress">
    <Stories.onPress />
  </Story>
</Preview>

### selectable

When `false`, the text is not selectable.

<Preview withSource='none'>
  <Story name="selectable">
    <Stories.selectable />
  </Story>
</Preview>

### style

Text supports typographic styles.

```js
{
  ...ViewStyle,
  color: color,
  fontFamily: string,
  fontFeatureSettings: string,
  fontSize: number | string,
  fontStyle: string,
  fontVariant: Array<string>,
  fontWeight: string,
  letterSpacing: string,
  lineHeight: string,
  textAlign: "center" | "end" | "inherit" | "justify" | "justify-all" | "left" | "right" | "start",
  textAlignVertical: string,
  textDecorationColor: color,
  textDecorationLine: string,
  textDecorationStyle: string,
  textIndent: string,
  textOverflow: string,
  textRendering: string,
  textShadowColor: color,
  textShadowOffset: { height: number, width: number },
  textShadowRadius: number | string,
  textTransform: string,
  unicodeBidi: string,
  whiteSpace: string,
  wordBreak: "normal" | "break-all" | "break-word" | "keep-all",
  wordWrap: string,
  writingDirection: string
}
```

<Preview withSource='none'>
  <Story name="style">
    <Stories.style />
  </Story>
</Preview>
