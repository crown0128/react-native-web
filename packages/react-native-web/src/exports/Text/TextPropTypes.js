/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import StyleSheetPropType from '../../modules/StyleSheetPropType';
import TextStylePropTypes from './TextStylePropTypes';
import { any, bool, func, number, object, oneOf, string } from 'prop-types';

const TextPropTypes = {
  accessibilityLabel: string,
  accessibilityLiveRegion: oneOf(['assertive', 'none', 'polite']),
  accessibilityRelationship: object,
  accessibilityRole: oneOf([
    'button',
    'header',
    'heading',
    'label',
    'link',
    'listitem',
    'none',
    'text'
  ]),
  accessible: bool,
  accessibilityState: object,
  children: any,
  importantForAccessibility: oneOf(['auto', 'no', 'no-hide-descendants', 'yes']),
  maxFontSizeMultiplier: number,
  nativeID: string,
  numberOfLines: number,
  onBlur: func,
  onFocus: func,
  onLayout: func,
  onPress: func,
  selectable: bool,
  style: StyleSheetPropType(TextStylePropTypes),
  testID: string,
  // web extensions
  onContextMenu: func,
  itemID: string,
  itemRef: string,
  itemProp: string,
  itemScope: string,
  itemType: string
};

export default TextPropTypes;
