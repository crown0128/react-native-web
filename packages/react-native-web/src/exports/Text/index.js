/**
 * Copyright (c) Nicolas Gallagher.
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import type { TextProps } from './types';

import createElement from '../createElement';
import css from '../StyleSheet/css';
import setAndForwardRef from '../../modules/setAndForwardRef';
import useElementLayout from '../../hooks/useElementLayout';
import usePlatformMethods from '../../hooks/usePlatformMethods';
import useResponderEvents from '../../hooks/useResponderEvents';
import React, { forwardRef, useContext, useRef } from 'react';
import StyleSheet from '../StyleSheet';
import TextAncestorContext from './TextAncestorContext';

const Text = forwardRef<TextProps, *>((props, ref) => {
  const {
    accessibilityLabel,
    accessibilityLiveRegion,
    accessibilityRelationship,
    accessibilityRole,
    accessibilityState,
    children,
    dir,
    forwardedRef,
    importantForAccessibility,
    lang,
    nativeID,
    numberOfLines,
    onBlur,
    onClick,
    onContextMenu,
    onFocus,
    onLayout,
    onPress,
    onMoveShouldSetResponder,
    onMoveShouldSetResponderCapture,
    onResponderEnd,
    onResponderGrant,
    onResponderMove,
    onResponderReject,
    onResponderRelease,
    onResponderStart,
    onResponderTerminate,
    onResponderTerminationRequest,
    onScrollShouldSetResponder,
    onScrollShouldSetResponderCapture,
    onSelectionChangeShouldSetResponder,
    onSelectionChangeShouldSetResponderCapture,
    onStartShouldSetResponder,
    onStartShouldSetResponderCapture,
    selectable,
    testID,
    // unstable
    onMouseDown,
    onMouseEnter,
    onMouseLeave,
    onMouseMove,
    onMouseOver,
    onMouseOut,
    onMouseUp,
    onTouchCancel,
    onTouchCancelCapture,
    onTouchEnd,
    onTouchEndCapture,
    onTouchMove,
    onTouchMoveCapture,
    onTouchStart,
    onTouchStartCapture,
    href,
    itemID,
    itemRef,
    itemProp,
    itemScope,
    itemType,
    rel,
    target,
    unstable_ariaSet,
    unstable_dataSet
  } = props;

  const hasTextAncestor = useContext(TextAncestorContext);
  const hostRef = useRef(null);
  const setRef = setAndForwardRef({
    getForwardedRef: () => forwardedRef,
    setLocalRef: c => {
      hostRef.current = c;
    }
  });

  const classList = [
    classes.text,
    hasTextAncestor === true && classes.textHasAncestor,
    numberOfLines === 1 && classes.textOneLine,
    numberOfLines != null && numberOfLines > 1 && classes.textMultiLine
  ];
  const style = [
    props.style,
    numberOfLines != null && numberOfLines > 1 && { WebkitLineClamp: numberOfLines },
    selectable === false && styles.notSelectable,
    onPress && styles.pressable
  ];

  useElementLayout(hostRef, onLayout);
  usePlatformMethods(hostRef, ref, classList, style);
  useResponderEvents(hostRef, {
    onMoveShouldSetResponder,
    onMoveShouldSetResponderCapture,
    onResponderEnd,
    onResponderGrant,
    onResponderMove,
    onResponderReject,
    onResponderRelease,
    onResponderStart,
    onResponderTerminate,
    onResponderTerminationRequest,
    onScrollShouldSetResponder,
    onScrollShouldSetResponderCapture,
    onSelectionChangeShouldSetResponder,
    onSelectionChangeShouldSetResponderCapture,
    onStartShouldSetResponder,
    onStartShouldSetResponderCapture
  });

  function handleClick(e) {
    if (onClick != null) {
      onClick(e);
    }
    if (onClick == null && onPress != null) {
      e.stopPropagation();
      onPress(e);
    }
  }

  const component = hasTextAncestor ? 'span' : 'div';
  const element = createElement(component, {
    accessibilityLabel,
    accessibilityLiveRegion,
    accessibilityRelationship,
    accessibilityRole,
    accessibilityState,
    accessible: onPress != null ? true : null,
    children,
    classList,
    // allow browsers to automatically infer the language writing direction
    dir: dir !== undefined ? dir : 'auto',
    importantForAccessibility,
    lang,
    nativeID,
    onBlur,
    onClick: handleClick,
    onContextMenu,
    onFocus,
    ref: setRef,
    style,
    testID,
    // unstable
    onMouseDown,
    onMouseEnter,
    onMouseLeave,
    onMouseMove,
    onMouseOver,
    onMouseOut,
    onMouseUp,
    onTouchCancel,
    onTouchCancelCapture,
    onTouchEnd,
    onTouchEndCapture,
    onTouchMove,
    onTouchMoveCapture,
    onTouchStart,
    onTouchStartCapture,
    href,
    itemID,
    itemRef,
    itemProp,
    itemScope,
    itemType,
    rel,
    target,
    unstable_ariaSet,
    unstable_dataSet
  });

  return hasTextAncestor ? (
    element
  ) : (
    <TextAncestorContext.Provider value={true}>{element}</TextAncestorContext.Provider>
  );
});

Text.displayName = 'Text';

const classes = css.create({
  text: {
    border: '0 solid black',
    boxSizing: 'border-box',
    color: 'black',
    display: 'inline',
    font: '14px System',
    margin: 0,
    padding: 0,
    whiteSpace: 'pre-wrap',
    wordWrap: 'break-word'
  },
  textHasAncestor: {
    color: 'inherit',
    font: 'inherit',
    whiteSpace: 'inherit'
  },
  textOneLine: {
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  },
  // See #13
  textMultiLine: {
    display: '-webkit-box',
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    WebkitBoxOrient: 'vertical'
  }
});

const styles = StyleSheet.create({
  notSelectable: {
    userSelect: 'none'
  },
  pressable: {
    cursor: 'pointer'
  }
});

export default Text;
