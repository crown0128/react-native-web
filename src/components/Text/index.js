import { pickProps } from '../../modules/filterObjectProps'
import CoreComponent from '../CoreComponent'
import React, { PropTypes } from 'react'
import TextStylePropTypes from './TextStylePropTypes'

const textStyleKeys = Object.keys(TextStylePropTypes)

const styles = {
  initial: {
    color: 'inherit',
    display: 'inline-block',
    font: 'inherit',
    margin: 0,
    padding: 0,
    wordWrap: 'break-word'
  },
  singleLineStyle: {
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  }
}

class Text extends React.Component {
  static propTypes = {
    accessibilityLabel: PropTypes.string,
    accessible: PropTypes.bool,
    children: PropTypes.any,
    component: CoreComponent.propTypes.component,
    numberOfLines: PropTypes.number,
    onPress: PropTypes.func,
    style: PropTypes.shape(TextStylePropTypes),
    testID: CoreComponent.propTypes.testID
  }

  static stylePropTypes = TextStylePropTypes

  static defaultProps = {
    component: 'span',
    style: styles.initial
  }

  _onPress(e) {
    if (this.props.onPress) this.props.onPress(e)
  }

  render() {
    const {
      accessibilityLabel,
      accessible,
      children,
      component,
      numberOfLines,
      onPress,
      style,
      testID,
      ...other
    } = this.props

    const resolvedStyle = pickProps(style, textStyleKeys)

    return (
      <CoreComponent
        {...other}
        aria-hidden={accessible ? null : true}
        aria-label={accessibilityLabel}
        children={children}
        className={'Text'}
        component={component}
        onClick={this._onPress.bind(this)}
        style={{
          ...(styles.initial),
          ...resolvedStyle,
          ...(numberOfLines === 1 && styles.singleLineStyle)
        }}
        testID={testID}
      />
    )
  }
}

export default Text
