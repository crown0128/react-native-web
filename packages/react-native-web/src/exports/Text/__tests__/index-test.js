/* eslint-env jasmine, jest */
/* eslint-disable react/jsx-no-bind */

import React from 'react';
import { render } from '@testing-library/react';
import Text from '../';

describe('components/Text', () => {
  test('default', () => {
    const { container } = render(<Text />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('nested', () => {
    const { container } = render(<Text children={<Text testID="child" />} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('allows "dir" to be overridden', () => {
    const { container } = render(<Text dir="rtl" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('prop "selectable"', () => {
    test('value of false', () => {
      const { container } = render(<Text selectable={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('value of true', () => {
      const { container } = render(<Text selectable={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
