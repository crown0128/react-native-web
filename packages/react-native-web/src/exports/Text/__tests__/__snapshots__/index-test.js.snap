// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Text prop "onPress" 1`] = `
<div
  className="rn-borderWidth-1yadl64 rn-boxSizing-deolkf rn-color-homxoj rn-cursor-1loqt21 rn-display-1471scf rn-fontFamily-1qd0xha rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-1kfwfc5 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-margin-crgep1 rn-padding-t60dpp rn-textDecorationLine-13wfysu rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-focusable={true}
  dir="auto"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
/>
`;

exports[`components/Text prop "selectable" 1`] = `
<div
  className="rn-borderWidth-1yadl64 rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-1qd0xha rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-1kfwfc5 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-margin-crgep1 rn-padding-t60dpp rn-textDecorationLine-13wfysu rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
/>
`;

exports[`components/Text prop "selectable" 2`] = `
<div
  className="rn-borderWidth-1yadl64 rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-1qd0xha rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-1kfwfc5 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-margin-crgep1 rn-padding-t60dpp rn-textDecorationLine-13wfysu rn-userSelect-lrvibr rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
/>
`;
